torch/quantization/fx/quantization_patterns.py

```diff
@@ -170,60 +176,81 @@ def convert(self, quantizer: QuantizerCls, node: Node, load_arg: Callable,
-                scale, zero_point = activation_post_process.calculate_qparams()
-                scale = float(scale)
-                zero_point = int(zero_point)
-                scale_arg, zero_point_arg = create_qparam_nodes(quantizer, node.name, scale, zero_point)
+                return quantize_node(
+                    quantizer, op_out, activation_post_process,
+                    node, is_input=False)
+            else:
+                warnings.warn(
+                    "No implementation found for dtype combination: {}"
+                    "for op {} with is_reference={} despite it being listed as supported"
+                    "this should not happen".format(dtypes, self.binary_op, is_reference))
+                assert False
+        elif not is_reference and self.binary_op in binary_op_supported_dtypes and dtypes in binary_op_supported_dtypes[self.binary_op]:
+            if dtypes in [(torch.quint8, torch.qint8, None)]:
+                assert self.quantized_binary_op is not None
+                if self.num_tensor_args == 1:
+                    # add/mul scalar
+                    first_arg = self.binary_op_node.args[0]
+                    cache_for_no_tensor_check: Dict[Node, bool] = dict()
+                    if isinstance(first_arg, Node) and (
+                            not all_node_args_have_no_tensors(
+                                first_arg, quantizer.modules, cache_for_no_tensor_check)):
+                        quantized_index = 0
+                    else:
+                        quantized_index = 1
 
-                if self.relu_node is not None:
-                    op = torch.ops.quantized.add_relu
+                    return quantizer.quantized_graph.create_node(
+                        'call_function', self.quantized_binary_op,
+                        load_arg(quantized=[quantized_index])(self.binary_op_node.args), self.binary_op_node.kwargs)
                 else:
-                    op = torch.ops.quantized.add
-                kwargs = {**self.binary_op_node.kwargs}
-                add_args = (*load_arg(quantized=True)(self.binary_op_node.args), scale_arg, zero_point_arg)
-                op = quantizer.quantized_graph.create_node(
-                    'call_function', self.quantized_binary_op, add_args, kwargs)
-                return op
+                    cur_idx = quantizer.activation_post_process_indexes[node.name]
+                    activation_post_process = \
+                        quantizer.modules[quantizer.activation_post_process_map[node.name][cur_idx]]
+                    quantizer.activation_post_process_indexes[node.name] += 1
+                    scale, zero_point = activation_post_process.calculate_qparams()
+                    scale = float(scale)
+                    zero_point = int(zero_point)
+                    scale_arg, zero_point_arg = create_qparam_nodes(quantizer, node.name, scale, zero_point)
+
+                    if self.relu_node is not None:
```

Comment:
Why are we hard coding the op to add /add_relu here? Should this be deleted (cc @jerryzh168 )

Reply:
Yeah it seems redundant but this code is unchanged from before.

Reply:
yeah looks like this can be deleted @HDCharles please delete them